source/common/filesystem/filesystem_impl.cc

```diff
@@ -168,7 +168,7 @@ void FileImpl::doWrite(Buffer::Instance& buffer) {
 void FileImpl::flushThreadFunc() {
 
   while (true) {
-    std::unique_lock<std::mutex> flush_lock;
+    std::unique_ptr<Thread::LockGuard> flush_lock;
```

Comment:
Why do we need to heap allocate this? This seems not optimal.

Reply:
This is a good question; my suspicion is that std::unique_lock also heap-allocated, but I could be wrong.  In any case I think I'll revert this, as it really didn't help annotation in this case to use a Thread::LockGuard, as the annotation analysis can't punch through the unique_ptr, sadly.